runtime/router.go

```diff
@@ -250,30 +158,48 @@ func (router *HTTPRouter) Register(
 	)
 }
 
-func (router *HTTPRouter) handleNotFound(w http.ResponseWriter, r *http.Request) {
-	resFields := []zapcore.Field{
-		zap.Int(statusCodeZapName, 404),
-	}
-	writeLogs(router.gateway.Logger, logRequestFields(r), resFields)
+func (router *HTTPRouter) handleNotFound(
+	w http.ResponseWriter,
+	r *http.Request,
+) {
+	var startTime, finishTime time.Time
+	defer func() {
+		// Post-process logs & metrics
+		resFields := []zapcore.Field{
+			zap.Int(statusCodeZapName, http.StatusNotFound),
+		}
+		writeLogs(router.logger, logRequestFields(r), resFields)
+		router.metrics[notFound].CollectMetrics(startTime, finishTime, http.StatusNotFound)
+	}()
+
+	startTime = time.Now()
 	// TODO custom NotFound
-	// A NotFound request is not started...
-	// TODO: inc.finish()
 	http.NotFound(w, r)
+	finishTime = time.Now()
 }
 
 func (router *HTTPRouter) handleMethodNotAllowed(
-	w http.ResponseWriter, r *http.Request,
+	w http.ResponseWriter,
+	r *http.Request,
 ) {
-	resFields := []zapcore.Field{
-		zap.Int(statusCodeZapName, 405),
-	}
-	writeLogs(router.gateway.Logger, logRequestFields(r), resFields)
+	var startTime, finishTime time.Time
```

Comment:
would prefer if we just allocated an `NewServerHTTPRequest()` here and started it. Then we can finish the res associated with it.